### ISSUE TITLE
Add ServiceDiscovery tests

### ISSUE THREAD

**Issue by marcosnils** [topfreegames/pitaya#186]
On this commit https://github.com/topfreegames/pitaya/commit/150c31e3f5623a0c90f5db6fd653f2c6e63990ef#diff-58d209d5c5b507c8971bebe6de3c3598ef2337b624f2f9171c314197fcc91b1aR30 we've changed the public Interface `ServiceDiscovery` and no tests had to be changed. This means that the interface doesn't have tests from the public perspective. 

I guess we can either make this interface private or add corresponding tests here. Also, adding tests for the `false` case I believe are required since we're not testing that explicitly.

**Comment by marcosnils** [comment]
Thinking about this a bit more, seems like v1.1.8 has been incorrectly tagged since it's not a minor release as it breaks backwards compatibility on this API. I've checked our gitlab organization and seems like nobody seems to be using that function directly, however if we want to be "purists" about it, we should bump the major version for such change. 

Thoughts?

cc @leohahn

**Comment by felipejfc** [comment]
You're right! We should test it. We can't make the interface private because this would eliminate the possibility for people to implement their own ServiceDiscovery, and extensibility is one of pitaya's pillars.

I would like to bump the major, however, only if we rebase the V2 branch on top of master and then release the V2 at once!

**Comment by felipejfc** [comment]
Also, in my [sidecar branch](https://github.com/topfreegames/pitaya/tree/sidecar), I did some project structure refactoring to better fit into go's best practices and upgraded dependencies, I wouldn't like to bump the major again to release it, we would have v2 and v3 too close to each other, so lets merge everything together and launch v2.

**Comment by marcosnils** [comment]
> I wouldn't like to bump the major again to release it, we would have v2 and v3 too close to each other, so lets merge everything together and launch v2.

SGTM!

**Comment by leohahn** [comment]
Is the sync servers really necessary though? For example, pitaya itself does not use the method, so nothing would stop someone from implementing their own service discovery without it (we did that using the redis service discovery).
